docs/distributed_training.md

```diff
@@ -1,3 +1,475 @@
+
+#### Zero Code Change Hook Initialization
+
+If you are running your training script on a [DLC container](https://docs.aws.amazon.com/dlami/latest/devguide/deep-learning-containers-images.html).
+
+To do this, you will need to:
+
+1. Create a file with a hook specification.
+
+Here is a minimal hook specification.
+
+```
+vim smdebug_hook_config.json
+
+----
+{
+  "LocalPath": "/tmp/hvd_trial_data",
+  "HookParameters":{
+    "include_workers": "all",
+    "save_interval": 500,
+  }
+}
+```
+
+2. Set the following environment variable
+
+```
+export SMDEBUG_CONFIG_FILE_PATH=./smdebug_hook_config.json
+```
+
+
+### Analysis of saved data
+
+See [link](https://github.com/awslabs/sagemaker-debugger/blob/master/docs/analysis.md) for detailed description.
+
+- [Trial API](#trial-api)
+- [Tensor API](#tensor-api)
+
+
+#### Trial API
+
+For a full list of the Trial API, see [link](https://github.com/awslabs/sagemaker-debugger/blob/master/docs/analysis.md#trial-api)
+
+...
+##### workers
+
+Query for all the worker processes from which data was saved by smdebug during multi worker training.
+
+```
+trial.workers()
+```
+
+###### Returns
+`list[str]` A sorted list of names of worker processes from which data was saved. If using TensorFlow Mirrored Strategy for multi worker training, these represent names of different devices in the process. For Horovod, torch.distributed and similar distributed training approaches, these represent names of the form worker_0 where 0 is the rank of the process.
+
+
+#### Tensor API
+
+For a full list of the Trial API, see [link](https://github.com/awslabs/sagemaker-debugger/blob/master/docs/analysis.md#tensor-1)
+
+...
+##### workers
+
+Get all the workers for which this tensor was saved at a given step
+
+```python
+trial.tensor(name).workers(step_num, mode=modes.GLOBAL)
+```
+
+###### Arguments
+- `name (str)` The name of the tensor to be fetched
+- `step_num (int)` The step number whose value is to be returned for the mode passed through the next parameter.
+- `mode (smdebug.modes enum value)` The mode applicable for the step number passed above. Defaults to `modes.GLOBAL`
+
+###### Returns
+`list[str]` A list of worker names for which the tensor was saved at the given step.
+
+
+## Tensorflow 1
+
+### Horovod
+
+- [Custom Hook Initializaton](#custom-hook-initializaton)
+- [Zero Code Change Hook](#zero-code-change-hook)
+
+#### Custom Hook Initializaton
+
+See full [source-code](https://github.com/awslabs/sagemaker-debugger/blob/master/examples/tensorflow/local/horovod_mnist.py)
+
+Example of minimal hook configuration requried:
+
+```
+...
+    smd_hook = smd.SessionHook(
+        out_dir=args.out_dir,
+        save_config=smd.SaveConfig(save_interval=args.save_interval),
+        include_collections=["weights", "gradients"],
+        include_workers="one"
```

Comment:
link to what values can this have? 
Can you confirm what happens if I provide arbitrary string like "two" here.

Reply:
Currently the logic for include_workers is:

`self.save_all_workers = True if include_workers == "all" else False`